test/Control/Effect/Spec.hs

```diff
@@ -1,122 +0,0 @@
-{-# LANGUAGE FlexibleContexts, FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, MultiWayIf, TemplateHaskell, TypeApplications, TypeOperators, UndecidableInstances #-}
-{-# OPTIONS_GHC -O2 -fplugin Test.Inspection.Plugin #-}
-module Control.Effect.Spec
-( spec
-) where
-
-import Control.Carrier
-import Control.Carrier.Error.Either
-import Control.Carrier.Fail.Either
-import Control.Carrier.Reader
-import Control.Carrier.State.Strict
-import Control.Effect.Sum
-import Prelude hiding (fail)
-import Test.Hspec
-import Test.Inspection as Inspection
-
-spec :: Spec
-spec = do
-  inference
-  reinterpretation
-  interposition
-  fusion
-
-inference :: Spec
-inference = describe "inference" $ do
-  it "can be wrapped for better type inference" $
-    run (runHasEnv (runEnv "i" ((++) <$> askEnv <*> askEnv))) `shouldBe` "ii"
-
-askEnv :: Has (Reader env) sig m => HasEnv env m env
-askEnv = ask
-
-runEnv :: env -> HasEnv env (ReaderC env m) a -> HasEnv env m a
-runEnv r = HasEnv . runReader r . runHasEnv
-
-
-newtype HasEnv env m a = HasEnv { runHasEnv :: m a }
-  deriving (Applicative, Functor, Monad)
-
-instance Carrier sig m => Carrier sig (HasEnv env m) where
-  eff = HasEnv . eff . handleCoercible
-
-
-reinterpretation :: Spec
-reinterpretation = describe "reinterpretation" $ do
-  it "can reinterpret effects into other effects" $
```

Comment:
1. We’re testing this already in the actual library, e.g. with `FreshC`.

2. This is not a good example; it’s contrived, and ill-justified.

Therefore, I’ve opted to 🔥 it since it’s not serving any purpose here and I’d rather not spend the time rewriting it into an example; this is honestly deserving of a much better-thought-out example (cf #9).